### DISCUSSION TITLE
How external Fortran programs are programmed into mooseprojects

### DISCUSSION THREAD

**Discussion by lvkas521424** [idaholab/moose#28013] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello， everyone.
I am currently trying to get moose to compile one of my external Fortran programs, which has been modified to be compiled by gfortran as per your previous suggestion. However, I am currently experiencing some problems with the compilation process. How does compiling a Fortran program with moose generate the` .mod` file, when I see it compiling he reminds me of the missing ` .mod` file. So in order to compile successfully I have to copy the .mod compiled externally into it, which I think is a very stupid way, so I would like to ask you for help.
Here is the compilation script I wrote， SCAPN is  my external program.

![image](https://github.com/idaholab/moose/assets/166678573/bc3ea00f-c9a6-4445-bbd9-52f171dd79f6)

yours
lvjiahui

**Comment by GiudGiud** [comment]
@permcody @milljm 

I think you can find examples of compiling fortran programs in moose in the solid_mechanics module
look at examples/examples/uel_build_tests
https://github.com/idaholab/moose/tree/next/modules/solid_mechanics/examples/uel_build_tests

**Comment by permcody** [comment]
We need more details. Are you pulling in a few routines, or are you
planning on coupling in a complete project?
